jest.config.ts

```diff
@@ -0,0 +1,88 @@
+import type {Config} from "jest";
+import os from "node:os";
+import pathPosix from "node:path/posix";
+
+
+let PRIMARY_REPORTER: string | [string, Record<string, unknown>] = "default";
+if ("undefined" !== typeof process.env.CI) {
+    PRIMARY_REPORTER = [
+        "github-actions",
+        {silent: false},
+    ];
+}
+console.log(`Environment variable "CI"="${process.env.CI}": ` +
+    `primary reporter will be ${JSON.stringify(PRIMARY_REPORTER)}`);
+
+const JEST_CONFIG: Config = {
+    collectCoverage: true,
+    collectCoverageFrom: ["src/**/*.{ts,tsx}"],
+    coverageProvider: "v8",
+    coverageReporters: ["text"],
+    coverageThreshold: {
+        "global": {
+            functions: 90,
+            lines: 90,
+        },
```

Comment:
* How about setting it to 100% for everything and then add exceptions on a case-by-case basis?
* We should specify a target for `branches` coverage, right?

Reply:
* Sounds good.
* Any numbers in your mind?